src/visual.ts

```diff
@@ -810,6 +870,91 @@ module powerbi.extensibility.visual {
             return labelSettings;
         }
 
+        public static parseXAxisLabelsSettings(objects: IDataViewObjects): MekkoXAxisLabelsSettings {
+            const enableRotataion: boolean = DataViewObjects.getValue(
+                objects,
+                MekkoChart.Properties["xAxisLabels"]["enableRotataion"],
+                MekkoChart.DefaultSettings.xAxisLabels.enableRotataion);
+
+            return {
+                enableRotataion
+            };
+        }
+
+        public static parseDataPointSettings(objects: IDataViewObjects): MekkoDataPointSettings {
+            const categoryGradient: boolean = DataViewObjects.getValue(
+                objects,
+                MekkoChart.Properties["dataPoint"]["categoryGradient"],
+                MekkoChart.DefaultSettings.dataPoint.categoryGradient);
+
+            const colorGradientEndColor: string = DataViewObjects.getValue(
+                objects,
+                MekkoChart.Properties["dataPoint"]["colorGradientEndColor"],
+                MekkoChart.DefaultSettings.dataPoint.colorGradientEndColor);
+
+            const colorDistribution: boolean = DataViewObjects.getValue(
+                objects,
+                MekkoChart.Properties["dataPoint"]["colorDistribution"],
+                MekkoChart.DefaultSettings.dataPoint.colorDistribution);
+
+            return {
+                categoryGradient,
+                colorGradientEndColor,
+                colorDistribution
+            };
+        }
+        public static parseSeriesSortSettings(objects: IDataViewObjects): MekkoSeriesSortSettings {
+            const enabled: boolean = DataViewObjects.getValue(
+                objects,
+                MekkoChart.Properties["sortSeries"]["enabled"],
+                MekkoChart.DefaultSettings.sortSeries.enabled);
+
+            const direction: string = DataViewObjects.getValue(
+                objects,
+                MekkoChart.Properties["sortSeries"]["direction"],
+                MekkoChart.DefaultSettings.sortSeries.direction);
+
+            const displayPercents: string = DataViewObjects.getValue(
+                objects,
+                MekkoChart.Properties["sortSeries"]["displayPercents"],
+                MekkoChart.DefaultSettings.sortSeries.displayPercents);
+
+            return {
+                enabled,
+                direction,
+                displayPercents
+            };
+        }
+
+        public static parseLegendSortSettings(objects: IDataViewObjects): MekkoLegendSortSettings {
```

Comment:
@ignatvilesov  I think the better way it is - completely replacing all settings parser methods by DataViewObjectsParser. It will reduce the count of code row.  Now it's over 5000 rows.